agent/command/git.go

```diff
@@ -870,15 +871,29 @@ func (c *gitFetchProject) fetch(ctx context.Context,
 		}
 	}
 
-	// Clone the project's modules.
+	// We use context.Background() instead of ctx here because
+	// using ctx ruins the honeycomb traces (the require the same
+	// uncancelled context).
+	g, gCtx := errgroup.WithContext(context.Background())
+	g.SetLimit(10)
+
+	// Clone the project's modules in goroutines.
 	for _, moduleName := range conf.BuildVariant.Modules {
-		if err := ctx.Err(); err != nil {
-			return errors.Wrapf(err, "canceled while applying module '%s'", moduleName)
-		}
-		err = c.fetchModuleSource(ctx, comm, conf, logger, jpm, td, opts.token, opts.method, p, moduleName)
-		if err != nil {
-			return errors.Wrapf(err, "fetching module source '%s'", moduleName)
-		}
+		moduleName := moduleName
+		g.Go(func() error {
+			if err := ctx.Err(); err != nil {
+				return errors.Wrapf(err, "canceled while applying module '%s'", moduleName)
+			}
+			// If gCtx has an error, this goroutine should not run and the error will be processed in the Wait() call.
+			if err := gCtx.Err(); err != nil {
+				return nil
+			}
+			return errors.Wrapf(c.fetchModuleSource(ctx, comm, conf, logger, jpm, td, opts.token, opts.method, p, moduleName), "fetching module source '%s'", moduleName)
+		})
+	}
+
+	if err = g.Wait(); err != nil {
+		return errors.Wrap(err, "fetching project and module source")
```

Comment:
Will it be confusing to users if only some of the module clones are attempted?

Reply:
This is wrapped in a retry logic that should just retry fetching all the modules. our existing implementation just stops when any one module fails, so I don't think it is an expectation that we attempt every module- only that we fail and stop when we reach an error

Reply:
SGTM, thanks.